zerver/lib/message.py

```diff
@@ -546,6 +579,7 @@ def bulk_access_messages(
 
     subscribed_recipient_ids = set(get_subscribed_stream_recipient_ids_for_user(user_profile))
 
+    user_group_membership_details = UserGroupMembershipDetails(user_recursive_group_ids=None)
```

Comment:
Similarly, the `Stream` query here likely is missing a `select_related`. `base_bulk_get_user_queryset` might be a useful pattern to crib from -- we can have a `stream_for_content_access_queryset` function that is shared between the code paths here.

Reply:
I'm not sure if we will need the stream_for_content_access_queryset if we are not doing any select related.

Reply:
Yeah, it's only if we need that, and I think you're right that we don't.